controllers/operatorconfig_controller.go

```diff
@@ -54,51 +71,341 @@ type OperatorConfigReconciler struct {
-		return ctrl.Result{Requeue: true}, nil
-	} else {
-		log.Info("Proxy Configured")
+		return err
 	}
+	log.Info(fmt.Sprintf("creating %s/%s", obj.Kind, obj.Name))
+	if err := r.Create(ctx, &obj); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			log.Error(err, fmt.Sprintf("Failed to create %s", obj.Kind), "Namespace", obj.Namespace, "Name", obj.Name)
+			return err
+		}
+	}
+
+		tmp := v1.Deployment{}
+		if err := wait.Poll(time.Second*1, time.Second*15, func() (done bool, err error) {
+			if err := r.Get(ctx, types.NamespacedName{Namespace: obj.Namespace, Name: obj.Name}, &tmp); err != nil {
+				if errors.IsNotFound(err) {
+					return false, nil
+				} else {
+					return false, err
+				}
+			}
+			return true, nil
+		}); err != nil {
+			return err
+		}
 
-	return ctrl.Result{}, err
+	return nil
 }
 
-func configureProxy(ctx context.Context, log logr.Logger) error {
-	return createResourcesFromFile("proxy.yaml", ctx, log)
+func (r *OperatorConfigReconciler) createService(resource *unstructured.Unstructured, ctx context.Context, log logr.Logger, oc *cranev1alpha1.OperatorConfig) error {
+
+	var obj corev1.Service
+	err := runtime.DefaultUnstructuredConverter.
+		FromUnstructured(resource.UnstructuredContent(), &obj)
+	if err != nil {
+		return err
+	}
+
+	err = controllerutil.SetControllerReference(oc, &obj, r.Scheme)
+	if err != nil {
+		return err
+	}
+
+	log.Info(fmt.Sprintf("creating %s/%s", obj.Kind, obj.Name))
+	if err := r.Create(ctx, &obj); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			log.Error(err, fmt.Sprintf("Failed to create %s", obj.Kind), "Namespace", obj.Namespace, "Name", obj.Name)
+			return err
+		}
+	}
+
+	return nil
 }
 
-func configureCranePlugin(ctx context.Context, log logr.Logger) error {
-	return createResourcesFromFile("crane-ui-plugin.yaml", ctx, log)
+func (r *OperatorConfigReconciler) createConfigMap(resource *unstructured.Unstructured, ctx context.Context, log logr.Logger, oc *cranev1alpha1.OperatorConfig) error {
+
+	var obj corev1.ConfigMap
+	err := runtime.DefaultUnstructuredConverter.
+		FromUnstructured(resource.UnstructuredContent(), &obj)
+	if err != nil {
+		return err
+	}
+
+	err = controllerutil.SetControllerReference(oc, &obj, r.Scheme)
+	if err != nil {
+		return err
+	}
+
+	log.Info(fmt.Sprintf("creating %s/%s", obj.Kind, obj.Name))
+	if err := r.Create(ctx, &obj); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			log.Error(err, fmt.Sprintf("Failed to create %s", obj.Kind), "Namespace", obj.Namespace, "Name", obj.Name)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *OperatorConfigReconciler) createClusterTask(resource *unstructured.Unstructured, ctx context.Context, log logr.Logger, oc *cranev1alpha1.OperatorConfig) error {
+
+	var obj v1beta1.ClusterTask
+	err := runtime.DefaultUnstructuredConverter.
+		FromUnstructured(resource.UnstructuredContent(), &obj)
+	if err != nil {
+		return err
+	}
+
+	err = controllerutil.SetControllerReference(oc, &obj, r.Scheme)
```

Comment:
shouldnt be setting owner references to cluster scoped resources. We need to use finalizers here.

When we create a clusterscoped object, create it with a finalizer as well as label selector that can be used to filter all the cluster tasks created for this operatorconfig object (to avoid problems like 2 different operatorconfig CR reconciling on the same clustertasks at cluster scope)